doc/man3/OSSL_DECODER_CTX_new_by_EVP_PKEY.pod

```diff
@@ -36,23 +38,27 @@ OSSL_DECODER_CTX_set_passphrase_cb
 OSSL_DECODER_CTX_new_by_EVP_PKEY() is a utility function that creates a
 B<OSSL_DECODER_CTX>, finds all applicable decoder implementations and sets
 them up, so all the caller has to do next is call functions like
-OSSL_DECODE_from_bio().
+L<OSSL_DECODER_from_bio(3)>.  The caller may used the optional I<input_type>,
+I<input_struct>, I<keytype> and I<selection> to specify what the input is
+expected to contain.
 
 Internally OSSL_DECODER_CTX_new_by_EVP_PKEY() searches for all available
 L<EVP_KEYMGMT(3)> implementations, and then builds a list of all potential
 decoder implementations that may be able to process the encoded input into
 data suitable for B<EVP_PKEY>s.  All these implementations are implicitly
 fetched using I<libctx> and I<propquery>.
 
-The search of decoder implementations can be limited with I<input_type>,
-which specifies a starting input type.  NULL is valid input and signifies
-that the decoder implementations will find out the input type on their own.
-This is further explained in L<OSSL_DECODER_CTX_set_input_type(3)>.
-
-The search of decoder implementations can also be limited with I<keytype>,
-which specifies the expected resulting keytype.  NULL is valid input and
-signifies that the decoder implementations will find out the keytype on
-their own from the input they get.
+The search of decoder implementations can be limited with I<input_type> and
+I<input_struct> which specifies a starting input type and input structure.
+NULL is valid for both of them and signifies that the decoder implementations
+will find out the input type on their own.
+This is further explained in L<OSSL_DECODER_CTX_set_input_type(3)>
+and L<OSSL_DECODER_CTX_set_input_structure(3)>.
```

Comment:
How can the reader find out what valid values are for the input structure? The above link doesn't give any clue. Similarly for the input_type.

Reply:
I took a stab at this, and included the start of openssl-glossary(7)